pkg/internal/token/execCredentialPlugin.go

```diff
@@ -35,6 +37,9 @@ func New(o *Options) (ExecCredentialPlugin, error) {
 		cachedRecord: &defaultCachedRecordProvider{
 			file: o.authRecordCacheFile,
 		},
+		msalCachedRecord: &defaultMSALCacheProvider{
+			file: o.authRecordCacheFile,
```

Comment:
authentication record isn't meant to cache msal token.

https://github.com/Azure/azure-sdk-for-go/blob/aae2fb75ffccafc669db72bebc3c1a66332f48d7/sdk/azidentity/authentication_record.go#L21C72-L22C22